### ISSUE TITLE
Enhance ADO Deposit. Add a JSON:API representation and also Revisions

### ISSUE THREAD

**Issue by DiegoPino** [esmero/strawberryfield#77]
Related to #37 

Currently we deposit only one version of the Object. And we deposit RAW JSON and a full entity serialization. 

* Let's deposit every revision (and add a setting where people can decide if they want to delete this files when removing ADOs or not). Its just a matter of using `\Drupal\strawberryfield\EventSubscriber\StrawberryfieldEventInsertSubscriberDepositDO` in a class that extends `\Drupal\strawberryfield\EventSubscriber\StrawberryfieldEventNewrevisionSubscriber` and add the Revision id. to the file name. As simple as that.

* But most important ,use the JSON:API extras approach to call the JSON:API serializer directly. See
https://git.drupalcode.org/project/jsonapi_extras/blob/8.x-3.x/src/EntityToJsonApi.php
Basically its just a wrapper, but a handy one that returns a RAW string. We can use that module (and make SBF depend on JSON:API and this one too, because, well, we like it and see that D9/10 will go that route fully also)

This is a super easy task, in case someone wants it.
